airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java

```diff
@@ -0,0 +1,205 @@
+  }
+
+  public abstract JsonNode toJdbcConfig(JsonNode config);
+
+  @Override
+  public ConnectorSpecification spec() throws IOException {
+    // return a JsonSchema representation of the spec for the integration.
+    final String resourceString = MoreResources.readResource("spec.json");
+    return Jsons.deserialize(resourceString, ConnectorSpecification.class);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try (final Database database = getDatabase(config)) {
+      // attempt to get current schema. this is a cheap query to sanity check that we can connect to the
+      // database. `currentSchema()` is a jooq method that will run the appropriate query based on which
+      // database it is connected to.
+      database.query(this::getCurrentDatabaseName);
+
+      return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
+    } catch (Exception e) {
+      LOGGER.debug("Exception while checking connection: ", e);
+      return new AirbyteConnectionStatus()
+          .withStatus(Status.FAILED)
+          .withMessage("Can't connect with provided configuration.");
+    }
+  }
+
+  protected Database getDatabase(JsonNode config) {
+    final JsonNode jdbcConfig = toJdbcConfig(config);
+
+    return Databases.createDatabase(
+        jdbcConfig.get("username").asText(),
+        jdbcConfig.has("password") ? jdbcConfig.get("password").asText() : null,
+        jdbcConfig.get("jdbc_url").asText(),
+        driverClass,
+        dialect);
+  }
+
+  protected String getCurrentDatabaseName(DSLContext ctx) {
+    return ctx.select(currentSchema()).fetch().get(0).getValue(0, String.class);
+  }
+
+  @Override
+  public SQLNamingResolvable getNamingResolver() {
+    return namingResolver;
+  }
+
+  /**
+   * Return the default schema where to data in the destination if the catalog doesn't specify any
+   * location
+   *
+   * @param config The destination configuration
+   * @return default schema name where to save data
+   */
+  protected String getDefaultSchemaName(JsonNode config) {
+    if (config.has("schema")) {
+      return config.get("schema").asText();
+    } else {
+      return "public";
+    }
+  }
+
+  @Override
+  protected void connectDatabase(JsonNode config) {
+    databaseConnection = getDatabase(config);
+  }
+
+  protected Database getDatabaseConnection() {
+    return databaseConnection;
+  }
+
+  @Override
+  public void queryDatabase(String query) throws SQLException {
+    getDatabaseConnection().query(ctx -> ctx.execute(query));
+  }
+
+  @Override
+  public void queryDatabaseInTransaction(String queries) throws Exception {
+    getDatabaseConnection().transaction(ctx -> ctx.execute(queries));
+  }
+
+  @Override
+  public String createRawTableQuery(String schemaName, String streamName) {
+    return String.format(
+        "CREATE TABLE IF NOT EXISTS %s.%s ( \n"
+            + "ab_id VARCHAR PRIMARY KEY,\n"
+            + "%s JSONB,\n"
```

Comment:
i think this is postgres specific? i think i would prefer if this were just left as abstract that the extender _has_ to override. otherwise if you're using a db that doesn't support this, you're not going to find out until runtime. (the abstract jdbc source may do the the same thing, which is because at the time we thought jooq was going to solve all of our problems but then it didn't). what do you think?

Reply:
Yes you are right, we could move it down to postgres destination instead